python/ray/serve/tests/test_batching.py

```diff
@@ -214,6 +215,36 @@ async def call(arg):
         await t3
 
 
+@pytest.mark.asyncio
+async def test_batch_timeout_empty_queue():
+    """Check that Serve waits for before creating the first batch."""
+
+    @serve.batch(max_batch_size=10, batch_wait_timeout_s=0.1)
+    async def no_op(requests):
+        return ["No-op"] * len(requests)
+
+    # Run one request, so internal batch queue gets constructed
+    await get_or_create_event_loop().create_task(no_op(None))
+
+    time.sleep(0.1)
```

Comment:
this needs to await to not block loop

Reply:
also why do we need it -- preferable to just `await` the task created above

Reply:
The `time.sleep(0.1)` is necessary to actually test the behavior. Without the fix in this change, the queue needs to be empty for `batch_wait_timeout_s` seconds for `@serve.batch` to just process the first request that shows up.

Reply:
> this needs to await to not block loop

What loop is it blocking?

Reply:
The event loop -- the `@serve.batch` code isn't running at all because you didn't `await`

Reply:
> The time.sleep(0.1) is necessary to actually test the behavior. Without the fix in this change, the queue needs to be empty for batch_wait_timeout_s seconds for @serve.batch to just process the first request that shows up.

Can't you just `await` that first request instead of the sleep?

Reply:
I think this is resolved by [this comment](https://github.com/ray-project/ray/pull/36510#discussion_r1232842594). Let me know if it's not.